### PR TITLE
Reorganizing Introduction to Concepts

### PR DESCRIPTION
**Pull request by ned1313**
This portion of the reorganization will involve the renaming of the Introduction section to Concepts. The concepts will include new sections that explain some basic concepts and then point to guides for examples and implementation details.

### CODE REVIEW

File: docs/content/introduction/security/_index.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+Some of the security features are only available on the Enterprise version of Gloo. They have been marked as such where applicable.
+
+---
+
+## External Authentication
+
+API Gateways act as a control point for the outside world to access the various application running in your environment. These application need to accept incoming requests from external end users. The incoming requests can be treated as anonymous or authenticated and depending on the requirements of your application. External authentication provides you with the ability to establish and validate who the client is, the service they are requesting, and define access or traffic control policies.
+
+External authentication is a Gloo Enterprise feature. It is possible to implement a [custom authentication server] when using the open-source version of Gloo.
```
Comment (kdorosh) [inline docs/content/introduction/security/_index.md]: is there a missing link here to a custom auth server?

Comment (kdorosh) [reply docs/content/introduction/security/_index.md]: Calling external auth an enterprise feature is misleading.

We expose external auth configuration in open-source if you provide your own envoy grpc service implementation of an external auth server.

Our implementation of this server is enterprise-only.

Comment (ned1313) [reply docs/content/introduction/security/_index.md]: This is still a work in progress. Links will be added shortly.

File: docs/content/introduction/security/_index.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+---
+
+## Network Encryption
+
+An API gateway sits between the downstream client and the upstream service it wants to connect with. The network traffic between the API gateway and the downstream client, and between the API gateway and the upstream service should be encrypted using Transport Layer Security (TLS). Gloo Gateway acts as the control plane, configuring Envoy through the xDS protocol. Ideally, that network traffic should also be encrypted through mutual TLS (mTLS).
+
+Mutual TLS requires that both the client and server present valid and trusted certificates when creating the TLS tunnel. Server-side TLS only requires that the server present a valid and trusted certificate to the client.
+
+Gloo is capable of configuring [server TLS] with downstream clients, [client TLS] with upstream services, and [mTLS between Gloo and Envoy].
```
Comment (kdorosh) [inline docs/content/introduction/security/_index.md]: more missing links

**Comment by ned1313** [comment]
Docs updated with links, content, and checked for broken links in the rest of docs.

**Pull request merged**